Sources/EthereumProvider/EthereumProvider.swift

```diff
@@ -0,0 +1,172 @@
+import BigInt
+import Foundation
+import curveSecp256k1
+import web3
+import mpc_core_kit_swift
+
+enum CustomSigningError: Error {
+    case generalError(error: String = "")
+
+    public var errorDescription: String {
+        switch self {
+        case let .generalError(err):
+            return err
+        }
+    }
+}
+
+enum EthereumSignerError: Error {
+    case emptyRawTransaction
+    case unknownError
+}
+
+extension  MpcCoreKit : EthereumAccountProtocol {
+    public var address: web3.EthereumAddress {
+        // try async
+        return EthereumAddress(KeyUtil.generateAddress(from: self.getTssPubKey().suffix(64) ).toChecksumAddress())
```

Comment:
Hmm. I remmeber self.getTssPubKey being both async and throwable, if I remember correctly it needs to remain so. What changed here?